language/dotnet/build-images.md

```diff
@@ -0,0 +1,307 @@
+```
+
+Docker images can be inherited from other images. Therefore, instead of creating our own image, we’ll use the official .NET SDK image that already has all the tools and packages that we need to build a .NET application.
+
+We will use a multi-stage build and define a stage for building the application.We define a `build-env` stage in our Dockerfile using `as`.
+
+> **Note**
+>
+> To learn more about multi-stage builds, see [Use multi-stage builds](../../develop/develop-images/multistage-build.md).
+
+To make things easier when running the rest of our commands, let’s create a working directory for our source files. This instructs Docker to use this path as the default location for all subsequent commands. By doing this, we do not have to type out full file paths but can use relative paths based on the working directory.
+
+```dockerfile
+WORKDIR /src
+```
+
+Although not necessary, the commands below will copy only the csproj files and then run `dotnet restore`. Each command creates a new container layer. To speed the building of containers, Docker caches these layers. Since these files won't change often, we can take advantage of the caching by copying these files and running restore as separate commands.
+
+
+```dockerfile
+COPY src/*.csproj .
+RUN dotnet restore
+```
+
+Next, you'll need to copy the rest of your source files into the image. The line below will copy the files from the `src` directory on your local machine to a directory called `src` in the image.
+
+```dockerfile
+COPY src ./src
+```
+
+Next, you'll need to run the `dotnet publish` command to build the project.
+
+```dockerfile
+RUN dotnet publish -c Release -o /publish
+```
+
+Next, you'll specify the image that you'll use to run the application, and define it as the `runtime` stage.
+
+```dockerfile
+FROM mcr.microsoft.com/dotnet/aspnet:6.0 as runtime
+```
+
+Next, specify the working directory for this stage.
+
+```dockerfile
+WORKDIR /app
+```
+
+Next, copy the /publish directory from the build-env stage into the runtime image.
+
+```dockerfile
+COPY --from=build-env /publish .
+```
+
+Expose port 80 to incoming requests.
+
+```dockerfile
+EXPOSE 80
+```
+
+Now, all we have to do is to tell Docker what command we want to run when our image is executed inside a container. We do this using the ENTRYPOINT command.
+
+```dockerfile
+ENTRYPOINT ["dotnet", "myWebApp.dll"]
+```
+
+Here's the complete Dockerfile.
+
+```dockerfile
+# syntax=docker/dockerfile:1
+
+FROM mcr.microsoft.com/dotnet/sdk:6.0 as build-env
+WORKDIR /src
+COPY src/*.csproj .
+RUN dotnet restore
+COPY src .
+RUN dotnet publish -c Release -o /publish
+
+FROM mcr.microsoft.com/dotnet/aspnet:6.0 as runtime
+WORKDIR /publish
+COPY --from=build-env /publish .
+EXPOSE 80
+ENTRYPOINT ["dotnet", "myWebApp.dll"]
+```
+
+## .dockerignore file
+
+To make your build context as small as possible, add a .dockerignore file to your `dotnet-docker` folder and copy the following into it.
+
+```console
+bin/
+obj/
```

Comment:
I got an error during the docker build. Running the application on the host created the `bin` and `obj` folders inside the src folder. So to ignore any `bin` and `obj` folder in any subdirectory, prepend the `**/`.

```suggestion
**/bin/
**/obj/
```

Reply:
Thank you. Will update.